src/stores/availableWorkflows.ts

```
@@ -1,54 +1,67 @@
+  () => {
+    const workflows = ref<WorkflowItem[]>([])
+    const preferredWorkflowIds = ref<string[]>([])
+
+    const workflowIds = computed(() => {
+      return workflows.value.map((workflow) => workflow.id)
+    })
+
+    const whatIfWorkflows = computed(() => {
+      return workflows.value.filter(
         (workflow) => workflow.whatIfTemplateId !== undefined,
       )
-    },
-  },
-  actions: {
-    byId(workflowId: string) {
-      const workflow = this.workflows.find(
+    })
+
+    function byId(workflowId: string) {
+      const workflow = workflows.value.find(
         (workflow) => workflow.id === workflowId,
       )
       return workflow
-    },
-    hasWhatIfTemplate(workflowId: string): boolean {
-      return this.byId(workflowId)?.whatIfTemplateId !== undefined
-    },
-    async fetch() {
-      this.workflows = await fetchWorkflowsWithExpectedRunTime()
-    },
-    setPreferredWorkflowIds(workflowIds: string[]): void {
+    }
+
+    function hasWhatIfTemplate(workflowId: string): boolean {
+      return byId(workflowId)?.whatIfTemplateId !== undefined
+    }
+
+    async function fetch() {
+      workflows.value = await fetchWorkflowsWithExpectedRunTime()
+    }
+
+    function setPreferredWorkflowIds(workflowIds: string[]): void {
       // Make sure the workflow IDs are unique.
       const uniqueWorkflowIds = uniq(workflowIds)
       // Check whether all workflows exist by call the byId function for all of
       // them; this should error if the workflow does not exist.
-      uniqueWorkflowIds.forEach((workflowId) => this.byId(workflowId))
-      this.preferredWorkflowIds = uniqueWorkflowIds
-    },
-    clearPreferredWorkflowIds(): void {
-      this.preferredWorkflowIds = []
-    },
+      uniqueWorkflowIds.forEach((workflowId) => byId(workflowId))
+      preferredWorkflowIds.value = uniqueWorkflowIds
+    }
+
+    function clearPreferredWorkflowIds(): void {
+      preferredWorkflowIds.value = []
+    }
+
+    fetch()
```

This will silently fail when an error is thrown, maybe at a `.catch`?